api/errors/errors.go

```diff
@@ -1,9 +1,30 @@
 package errors
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+	"math/big"
+)
 
 var (
 	ErrNotSupported = errors.New("endpoint is not supported")
 	ErrInvalid      = errors.New("invalid request")
 	ErrInternal     = errors.New("internal error")
 )
+
+type ErrGasPriceTooLow struct {
```

Comment:
nice!
btw at some later point I feel we should consolidate errors from different places. I think I made a mistake of having multiple places where we define errors.